### PR TITLE
added login, main(table and form) and registration page, deletedialog…

### PR DESCRIPTION
**Pull request by gmatyasheva**
…, route, protected route, styles

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import Header from "./components/Header/Header";
+import Register from "./components/Register/Register";
+import MainPage from "./components/MainPage/MainPage";
+import LogIn from "./components/LogIn/LogIn";
+import ProtectedRoute from "./ProtectedRoute/ProtectedRoute";
+import "./App.scss";
+
+const App = () => {
+  const isAuth = () => {
+    const token = JSON.parse(localStorage.getItem("token"));
```
Comment (usatovaDasha) [inline src/App.js:13]: it is strong. No need parse

File: src/ProtectedRoute/ProtectedRoute.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Redirect, Route } from "react-router-dom";
+
+function ProtectedRoute({ component: Component, ...restOfProps }) {
+  const token = JSON.parse(localStorage.getItem("token"));
```
Comment (usatovaDasha) [inline src/ProtectedRoute/ProtectedRoute.js:5]: higher

File: src/components/DeleteDialog/DeleteDialog.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import Button from "@material-ui/core/Button";
```
Comment (usatovaDasha) [inline src/components/DeleteDialog/DeleteDialog.js:2]: in one import core

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  });
+
+  return (
+    <header className="header">
+      <Container>
+        <div className="header-content">
+          <img src={logo} alt="logo"></img>
+          <h1>{title}</h1>
+        </div>
+        {location.pathname === "/main" ? (
```
Comment (usatovaDasha) [inline src/components/Header/Header.js:34]: location.pathname === "/main" && <Button

File: src/components/Header/Header.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          <Button
+            onClick={() => {
+              localStorage.removeItem("token");
```
Comment (usatovaDasha) [inline src/components/Header/Header.js:37]: in function

File: src/components/LogIn/LogIn.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useForm, Controller } from "react-hook-form";
+import axios from "axios";
+import { Button, TextField, Link, Container } from "@material-ui/core";
```
Comment (usatovaDasha) [inline src/components/LogIn/LogIn.js:5]: move to single lines

File: src/components/LogIn/LogIn.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+                render={({ field }) => (
+                  <TextField
+                    variant="outlined"
+                    {...field}
+                    placeholder="login"
+                    type="text"
+                  />
+                )}
+              />
+              {errors?.login && (
```
Comment (usatovaDasha) [inline src/components/LogIn/LogIn.js:55]: you using typescript?

File: src/components/LogIn/LogIn.scss [hunk 8]
```diff
@@ -0,0 +1,106 @@
+.login-container {
+  height: 100vh;
+
+  .MuiContainer-maxWidthLg {
+    display: flex;
+    align-items: center;
+    height: 80%;
+    padding: 20px 20px;
```
Comment (usatovaDasha) [inline src/components/LogIn/LogIn.scss:8]: padding: 20px;

File: src/components/MainPage/MainPage.scss [hunk 9]
```diff
@@ -0,0 +1,21 @@
+.main-container {
+  height: 100vh;
+
+  .MuiContainer-maxWidthLg {
+    display: flex;
+    align-items: center;
+    padding: 20px 20px;
```
Comment (usatovaDasha) [inline src/components/MainPage/MainPage.scss:7]: higher

File: src/components/MainPage/TablePage/TablePage.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import axios from "axios";
+
```
Comment (usatovaDasha) [inline src/components/MainPage/TablePage/TablePage.js:3]: no need empty line

File: src/components/MainPage/TablePage/TablePage.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    // });
+  };
+
+  return (
+    <>
+      <TableContainer component={Paper}>
+        <Table aria-label="customized table">
+          <TableHead>
+            <TableRow>
+              <TableCell>Имя</TableCell>
```
Comment (usatovaDasha) [inline src/components/MainPage/TablePage/TablePage.js:45]: in map header cell

File: src/components/Register/Register.js [hunk 12]
```diff
@@ -0,0 +1,138 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useForm, Controller } from "react-hook-form";
+import axios from "axios";
+import { Button, TextField, Link, Container } from "@material-ui/core";
```
Comment (usatovaDasha) [inline src/components/Register/Register.js:5]: move to single lines

File: src/components/Register/Register.js [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+
+const Register = () => {
+  const [backendError, setBackendError] = useState("");
+  let history = useHistory();
+  const {
+    handleSubmit,
+    getValues,
+    control,
+    formState: { errors },
+  } = useForm();
```
Comment (usatovaDasha) [inline src/components/Register/Register.js:18]: add empty line

File: src/components/Register/Register.js [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+  const onSubmit = async (data) => {
+    axios
+      .post("http://localhost:8080/patients", data)
+      .then((res) => {
+        if (res.data.success) {
+          axios
+            .post("http://localhost:8080/logInPatient", data)
+            .then((res) => {
+              if (res.data.success) {
+                localStorage.setItem("token", JSON.stringify(res.data.token));
```
Comment (usatovaDasha) [inline src/components/Register/Register.js:28]: token is string

File: src/components/EditDialog/EditDialog.js [hunk 15]
```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import { useForm, Controller } from "react-hook-form";
+import axios from "axios";
+import { Button, TextField, Select, MenuItem } from "@material-ui/core";
```
Comment (usatovaDasha) [inline src/components/EditDialog/EditDialog.js:4]: 2 core imports?

File: src/components/EditDialog/EditDialog.js [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "@material-ui/core";
+
+import "./EditDialog.scss";
+
+const EditDialog = ({
+  handleCloseEditDialog,
```
Comment (usatovaDasha) [inline src/components/EditDialog/EditDialog.js:16]: destructurization in component, but not in params

File: src/components/EditDialog/EditDialog.scss [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+    padding: 20px 25px;
+    box-shadow: 4px 4px 4px 4px rgba(0, 0, 0, 0.25);
+  }
+
+  .MuiDialogActions-root {
+    padding: 30px;
+  }
+
+  .MuiDialog-paperScrollPaper {
+     width: 100vw;
```
Comment (usatovaDasha) [inline src/components/EditDialog/EditDialog.scss:31]: tabs

File: src/components/Header/Header.js [hunk 18]
```diff
@@ -24,23 +24,23 @@ const Header = () => {
...
         </div>
-        {location.pathname === "/main" ? (
-          <Button
-            onClick={() => {
-              localStorage.removeItem("token");
-              history.push("/login");
-            }}
-          >
+        {location.pathname === "/main" && (
+          <Button onClick={() => {logOut()}}>
```
Comment (usatovaDasha) [inline src/components/Header/Header.js:40]: onClick={() => logOut()}

File: src/components/MainPage/TablePage/TablePage.js [hunk 19]
```diff
@@ -14,59 +13,103 @@ import {
 import DeleteOutlineIcon from "@material-ui/icons/DeleteOutline";
 import EditIcon from "@material-ui/icons/Edit";
 import DeleteDialog from "../../DeleteDialog/DeleteDialog";
+import EditDialog from "../../EditDialog/EditDialog";
 
 import "./TablePage.scss";
 
-const TablePage = ({ receptions, setReceptions }) => {
-  const [openDeleteDialog, setOpenDeleteDialog] = React.useState(false);
+const TablePage = ({ receptions, setReceptions, doctors, setBackendError, tableHeader }) => {
```
Comment (usatovaDasha) [inline src/components/MainPage/TablePage/TablePage.js:20]: destructurization in component

File: src/components/MainPage/TablePage/TablePage.js [hunk 20]
```diff
@@ -14,59 +13,103 @@ import {
...
+  const handleCloseEditDialog = () => {
+    setOpenEditDialog(false);
+  };
+
+  const convertDate = (strVal) => {
+    if (!strVal) {
+      return "";
+    }
+    const arr = strVal.split("-");
+    return [arr[2], arr[1], arr[0]].join(".");
```
Comment (usatovaDasha) [inline src/components/MainPage/TablePage/TablePage.js:49]: return arr.join(".");

File: src/components/MainPage/TablePage/TablePage.js [hunk 21]
```diff
@@ -14,59 +13,103 @@ import {
...
+  };
+
   const deleteReceptionText = () => {
-    //  axios
-    // .delete(`http://localhost:8080/receptions/${receptions[index]._id}`)
-    // .then((res) => {
-    //   setReceptions(res.data.data);
-    // });
+    console.log(deleteIndex, "del");
```
Comment (usatovaDasha) [inline src/components/MainPage/TablePage/TablePage.js:53]: remove console.log

File: src/components/MainPage/TablePage/TablePage.js [hunk 22]
```diff
@@ -14,59 +13,103 @@ import {
...
           <TableHead>
             <TableRow>
-              <TableCell>Имя</TableCell>
-              <TableCell align="center">Врач</TableCell>
-              <TableCell align="center">Дата</TableCell>
-              <TableCell align="center">Жалобы</TableCell>
-              <TableCell align="center"></TableCell>
+              {tableHeader.map((value, index) => (
+                <TableCell
+                  value={value}
```
Comment (usatovaDasha) [inline src/components/MainPage/TablePage/TablePage.js:71]: value no need

File: src/components/MainPage/TablePage/TablePage.js [hunk 23]
```diff
@@ -14,59 +13,103 @@ import {
...
                     onClick={() => {
-                      handleClickOpenDeleteDialog();
+                      handleClickOpenDeleteDialog(index);
                     }}
                   >
                     <DeleteOutlineIcon />
                   </IconButton>
-                  <IconButton>
+                  <IconButton
+                    onClick={() => {
```
Comment (usatovaDasha) [inline src/components/MainPage/TablePage/TablePage.js:109]: () => handleClickOpenEditDialog(index) }

File: src/components/EditDialog/EditDialog.js [hunk 24]
```diff
@@ -67,7 +73,12 @@ const EditDialog = ({
                 name="doctor"
                 control={control}
                 render={({ field }) => (
-                  <Select type="text" {...field} variant="outlined">
+                  <Select
+                    type="text"
+                    {...field}
+                    variant="outlined"
+                    defaultValue={receptions[editIndex].doctor}
```
Comment (usatovaDasha) [inline src/components/EditDialog/EditDialog.js:80]: destructurization

File: src/components/MainPage/SortField/SortField.js [hunk 25]
```diff
@@ -14,25 +13,197 @@ import AddBoxIcon from "@material-ui/icons/AddBox";
...
+  };
+
+  //yyyy-mm-dd
+  const parseDate = (strVal) => {
+    const arr = strVal.split("-");
+    return new Date(arr[0], arr[1] - 1, arr[2]);
+  };
+
+  const sortReceptions = (receptionsParam, sortFieldParam, sortOrderParam) => {
+    const sortOrderLocal = sortMapOrders[sortOrderParam] == 0 ? -1 : 1;
```
Comment (usatovaDasha) [inline src/components/MainPage/SortField/SortField.js:64]: sortMapOrders[sortOrderParam]  ? 1 : -1;

File: src/components/MainPage/SortField/SortField.js [hunk 26]
```diff
@@ -14,25 +13,197 @@ import AddBoxIcon from "@material-ui/icons/AddBox";
...
+  };
+
+  const onRemoveFilterClick = () => {
+    setIsVisibleFilter(false);
+    setReceptions(initialReceptions);
+    setStartDate("");
+    setEndDate("");
+  };
+
+  if (!initialReceptions && receptions && receptions.length !== 0) {
```
Comment (usatovaDasha) [inline src/components/MainPage/SortField/SortField.js:126]: (!initialReceptions && receptions && receptions.length)

File: src/components/Register/Register.js (hunk 12 above)
Comment (usatovaDasha) [reply src/components/Register/Register.js:5]: !!

File: src/components/MainPage/TablePage/TablePage.js (hunk 20 above)
Comment (gmatyasheva) [reply src/components/MainPage/TablePage/TablePage.js]: incorrect comments. array should be in reverse order --arr.reverse().join(".")

File: src/components/MainPage/SortField/SortField.js (hunk 25 above)
Comment (gmatyasheva) [reply src/components/MainPage/SortField/SortField.js:64]: incorrect comment. this update will change logic in opposite way.  Do you mean sortMapOrders[sortOrderParam] ? -1 : 1?

**Pull request merged**